src/mscorlib/shared/System/Globalization/TimeSpanParse.cs

```diff
@@ -1230,7 +1248,7 @@ private static bool TryParseByFormat(ReadOnlySpan<char> input, ReadOnlySpan<char
                         tokenLen = DateTimeFormat.ParseRepeatPattern(format, i, ch);
                         if (tokenLen > 2 || seenHH || !ParseExactDigits(ref tokenizer, tokenLen, out hh))
                         {
-                            return result.SetFailure(ParseFailureKind.Format, nameof(SR.Format_InvalidString));
+                            return result.SetFailure(nameof(SR.Format_InvalidString));
```

Comment:
>nameof(SR.Format_InvalidString)); [](start = 53, length = 33)

this looks wrong to me. this need to be just SR.Format_InvalidString without nameof. I know the old code was doing that but this wrong. 

could you please fix the rest of the code doing that?

Reply:
I don't see why this is wrong. Can you elaborate?

Reply:
nameof(SR.Format_InvalidString) will produce the string "Format_InvalidString" and not the resource string value of SR.Format_InvalidString which is "Input string was not in a correct format."

Reply:
Yes, and that's what the code is intended to do.

Reply:
Why not passing SR.Format_InvalidString returned value to SetFailure and not having SetFailure to lookup the string by itself? this looks weird to me that we have a property that calling it do the job and instead we get its name and then try to look up the resource to get the value which we can just get by calling the property.

Reply:
and it is easy to get the code reader confused too as we did :-)

Reply:
I think it's done way originally because if you check the implementation of `SetFailure` it has a condition before actually getting the value for the passed in key:

    if (!_throwOnFailure)
    {
        return false;
    }

This way the more expensive lookup done by `InternalGetResourceString` (which uses a lock/Monitor) can be avoided. For example when doing `TryParse`.

Reply:
Looking at the code, it looks to me the majority of the calls to setFailure is used with the Format_InvalidString and Format_BadQuote, so you may create a helper methods for those as you did with SetOverflowFailure and SetBadTimeSpanFailure